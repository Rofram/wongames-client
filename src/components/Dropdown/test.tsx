import { screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { renderWithTheme } from 'utils/tests/helpers'

import Dropdown from '.'

describe('<Dropdown />', () => {
  it('should render the component correctly', () => {
    const { container } = renderWithTheme(
      <Dropdown title={<span>Click here</span>}>
        <span>content</span>
      </Dropdown>
    )

    const content = screen.getByText(/content/i).parentElement

    expect(screen.getByText(/click here/i)).toBeInTheDocument()

    expect(content).toHaveStyle({
      opacity: 0
    })

    expect(content).toHaveAttribute('aria-hidden', 'true')

    userEvent.click(screen.getByText(/click here/i))

    expect(content).toHaveStyle({
      opacity: 1
    })

    expect(content).toHaveAttribute('aria-hidden', 'false')

    expect(container.firstChild).toMatchSnapshot()
  })
})
