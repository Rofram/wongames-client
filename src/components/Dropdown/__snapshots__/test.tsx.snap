// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> should render the component correctly 1`] = `
.c1 {
  cursor: pointer;
  color: #FAFAFA;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 2.4rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: #FAFAFA;
  color: #030517;
  margin-top: 2.4rem;
  position: absolute;
  right: 0;
}

.c3::before {
  content: '';
  position: absolute;
  border-right: 1.2rem solid transparent;
  border-left: 1.2rem solid transparent;
  border-bottom: 1.2rem solid #FAFAFA;
  top: -1.2rem;
  right: 2.4rem;
}

.c4 .c2 {
  -webkit-transition: -webkit-transform 0.2s ease-in,opacity 0.3s ease-in-out;
  -webkit-transition: transform 0.2s ease-in,opacity 0.3s ease-in-out;
  transition: transform 0.2s ease-in,opacity 0.3s ease-in-out;
  opacity: 0;
  pointer-events: none;
  -webkit-transform: translateY(-2rem);
  -ms-transform: translateY(-2rem);
  transform: translateY(-2rem);
}

.c0 {
  position: relative;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
}

.c0 .c2 {
  -webkit-transition: -webkit-transform 0.2s ease-in,opacity 0.3s ease-in-out;
  -webkit-transition: transform 0.2s ease-in,opacity 0.3s ease-in-out;
  transition: transform 0.2s ease-in,opacity 0.3s ease-in-out;
  opacity: 1;
  pointer-events: auto;
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <span>
      Click here
    </span>
  </div>
  <div
    aria-hidden="false"
    class="c2 c3"
  >
    <span>
      content
    </span>
  </div>
</div>
`
